AFC.py

```diff
@@ -650,13 +653,17 @@ def cmd_CHANGE_TOOL(self, gcmd):
             self.toolhead.wait_moves()
             if self.is_printing() and not self.is_paused():
                 self.change_tool_pos = None
+                # Restore state
+                self.gcode.run_script_from_command("RESTORE_GCODE_STATE NAME=_AFC_CHANGE_TOOL MOVE=1 MOVE_SPEED={}".format(self.tool_unload_speed))
 
     cmd_RESTORE_CHANGE_TOOL_POS_help = "change filaments in tool head"
     def cmd_RESTORE_CHANGE_TOOL_POS(self, gcmd):
         if self.change_tool_pos:
             restore_pos = self.change_tool_pos[:3]
             self.toolhead.manual_move(restore_pos, self.tool_unload_speed)
             self.toolhead.wait_moves()
```

Comment:
We can probably remove these since the restore state below will do the move for us.

```suggestion
```

Reply:
If this is removed then I think it might be better to remove the save/restore out of the if checks in CHANGE_TOOL, otherwise there will be z movement without restoring when changing lanes outside of printing I think.